src/LinuxTracing/UprobesUnwindingVisitorDwarfUnwindingTest.cpp

```diff
@@ -0,0 +1,1814 @@
+              ::testing::Property(&orbit_grpc_protos::FullAddressInfo::module_name,
+                                  TestFixture::kTargetName)),
+          ::testing::AllOf(
+              ::testing::Property(&orbit_grpc_protos::FullAddressInfo::absolute_address,
+                                  TestFixture::kTargetAddress2),
+              ::testing::Property(&orbit_grpc_protos::FullAddressInfo::function_name,
+                                  TestFixture::kFunctionName2),
+              ::testing::Property(&orbit_grpc_protos::FullAddressInfo::offset_in_function, 0),
+              ::testing::Property(&orbit_grpc_protos::FullAddressInfo::module_name,
+                                  TestFixture::kTargetName)),
+          ::testing::AllOf(
+              ::testing::Property(&orbit_grpc_protos::FullAddressInfo::absolute_address,
+                                  TestFixture::kTargetAddress3),
+              ::testing::Property(&orbit_grpc_protos::FullAddressInfo::function_name,
+                                  TestFixture::kFunctionName3),
+              ::testing::Property(&orbit_grpc_protos::FullAddressInfo::offset_in_function, 0),
+              ::testing::Property(&orbit_grpc_protos::FullAddressInfo::module_name,
+                                  TestFixture::kTargetName))));
+
+  EXPECT_EQ(unwinding_errors, 0);
+  EXPECT_EQ(discarded_samples_in_uretprobes_counter, 0);
+}
+
+REGISTER_TYPED_TEST_SUITE_P(
+    UprobesUnwindingVisitorDwarfUnwindingTest,
+    VisitValidStackSampleWithoutUprobesSendsCompleteCallstackAndAddressInfos,
+    VisitEmptyStackSampleWithoutUprobesDoesNothing,
+    VisitInvalidStackSampleWithoutUprobesSendsUnwindingErrorAndAddressInfos,
+    VisitSingleFrameStackSampleInFunctionToStopAtSendsCompleteCallstackAndAddressInfos,
+    VisitSingleFrameStackSampleOutsideOfAnyFunctionToStopAtSendsUnwindingErrorAndAddressInfos,
+    VisitSingleFrameStackSampleWithoutUprobesSendsUnwindingErrorAndAddressInfos,
+    VisitStackSampleStoppedAtUprobesSendsPatchingFailedCallstack,
+    VisitStackSampleStoppedAtUserSpaceInstrumentationTrampolineSendsPatchingFailedCallstack,
+    VisitStackSampleUsesLatestUserSpaceCallstack,
+    VisitStackSampleUsesUserSpaceCallstackOnlyFromSameThread, VisitStackSampleUsesUserSpaceStack,
+    VisitStackSampleUsesUserStackMemoryFromAllStreamIds,
+    VisitStackSampleWithinUprobeSendsInUprobesCallstack,
+    VisitStackSampleWithinUserSpaceInstrumentationLibraryButNotTrampolineSendsCompleteCallstack,
+    VisitStackSampleWithinUserSpaceInstrumentationTrampolineAndLibrarySendsInUserSpaceInstrumentationCallstack,
+    VisitStackSampleWithinUserSpaceInstrumentationTrampolineSendsInUserSpaceInstrumentationCallstack,
+    VisitTwoValidStackSamplesSendsAddressInfosOnlyOnce,
+    VisitValidStackSampleWithNullptrMapInfosSendsCompleteCallstackAndAddressInfosWithoutModuleName);
+
+using TestTypes = ::testing::Types<
+    std::pair<StackSamplePerfEvent, orbit_grpc_protos::FullCallstackSample>,
```

Comment:
We use `std::pair` just to carry two types, and then use `first_type` and `second_type`. I think we should have our own type:

```
template <typename T, typename U>
struct DwarfUnwindingTestType {
  using PerfEventT = T;
  using ProducerCaptureEventT = U;
};
```
So that then you can write `TypeParam::PerfEventT` and `TypeParam::ProducerCaptureEventT`, and that's much clearer in my opinion.

Reply:
Much better. Thanks!